### PR TITLE
added a check to detect duplicate members

### PR DESCRIPTION
**Pull request by Saumya25mis**
This PR contains a check for detecting and preventing duplicate members to be added

Edit - We were seeing a lot of duplicate entries in audit results for JS language. We were previously adding a new entry for each usage of a variable in code - This PR makes a set of of each variable based off the identifier name, class name and file name - and then checks the set before adding a new entry in report to prevent duplicates

### CODE REVIEW

**Comment by pandurangpatil** [comment]
@Saumya25mis add the details in the PR description summarising the changes in this PR.

File: src/main/scala/ai/privado/audit/DataElementDiscovery.scala [hunk 1]
```diff
@@ -592,11 +592,14 @@ object DataElementDiscoveryJS {
             AuditReportConstants.AUDIT_EMPTY_CELL_VALUE
           )
           val ruleMemberInfo = taggedMemberInfo.getOrElse(key.fullName, new mutable.HashMap[String, String])
+          val addedMembers   = mutable.Set[String]()
+
           value.foreach {
             case (member: Member) => {
-              if (member.name.nonEmpty) {
+              val memberDetails = s"${key.fullName}${key.file.head.name}${member.name}"
```
Comment (HiteshMah-Jan) [inline src/main/scala/ai/privado/audit/DataElementDiscovery.scala:599]: can we rename these memberDetails, localDetails, identifierDetails to a more descriptive one like memberUniqueKey

**Review by HiteshMah-Jan — DISMISSED** [review]

File: src/main/scala/ai/privado/audit/DataElementDiscovery.scala [hunk 2]
```diff
@@ -627,46 +630,52 @@ object DataElementDiscoveryJS {
...
-                  getFileScoreJS(key.file.name.headOption.getOrElse(Constants.EMPTY), xtocpg),
-                  param.name,
-                  param.typeFullName,
-                  AuditReportConstants.AUDIT_NOT_CHECKED_VALUE,
-                  AuditReportConstants.AUDIT_EMPTY_CELL_VALUE,
-                  AuditReportConstants.AUDIT_EMPTY_CELL_VALUE,
-                  AuditReportConstants.AUDIT_EMPTY_CELL_VALUE,
-                  AuditReportConstants.AUDIT_EMPTY_CELL_VALUE
-                )
+              val paramUniqueKey = s"${key.fullName}${key.file.head.name}${param.name}"
```
Comment (khemrajrathore) [inline src/main/scala/ai/privado/audit/DataElementDiscovery.scala:633]: update `key.file.head.name` to `key.file.name.headOption.getOrElse(Constants.EMPTY)`, 

Similarly update for all other occurrences

File: src/main/scala/ai/privado/audit/DataElementDiscovery.scala [hunk 3]
```diff
@@ -627,46 +630,52 @@ object DataElementDiscoveryJS {
...
+              if (!addedMembers.contains(paramUniqueKey)) {
+                addedMembers.add(paramUniqueKey)
+                if (ruleMemberInfo.contains(param.name)) {
```
[the remark this answers is not in the document]
Comment (khemrajrathore) [reply src/main/scala/ai/privado/audit/DataElementDiscovery.scala:636]: Similarly update other occurrence

File: src/main/scala/ai/privado/audit/DataElementDiscovery.scala [hunk 4]
```diff
@@ -596,10 +596,11 @@ object DataElementDiscoveryJS {
...
           value.foreach {
             case (member: Member) => {
-              val memberUniqueKey = s"${key.fullName}${key.file.head.name}${member.name}"
-              if (member.name.nonEmpty && !addedMembers.contains(memberUniqueKey)) {
-                addedMembers.add(memberUniqueKey)
-                if (ruleMemberInfo.contains(member.name)) {
+              val memberUniqueKey =
+                s"${key.fullName}${key.file.name.headOption.getOrElse(Constants.EMPTY)}${member.name}"
+              if (member.name.nonEmpty) {
+                if (ruleMemberInfo.contains(member.name) && !addedMembers.contains(memberUniqueKey)) {
```
Comment (khemrajrathore) [inline src/main/scala/ai/privado/audit/DataElementDiscovery.scala:602]: @Saumya25mis The earlier version of handling if condition was correct, this new commit is introducing checking `!addedMembers.contains(memberUniqueKey)` twice

cc - @HiteshMah-Jan

**Review by khemrajrathore — APPROVED** [review]
LGTM

**Pull request merged**